### PR TITLE
Big updates on logging, warnings and exceptions 

### PR DESCRIPTION
**Pull request by CunliangGeng**
The main updates are:
1. Add global logger to output necessary information to stdout, and exceptions and errors to stderr.
2. Add warnings to record warnings in combination with logger, by defalut output to stderr.
3. Refactor some code, see detail below.

Note: 
- The results of `test` generated by these updates are consistent with those by the old code.
- The `learn` code are not touched this time.

Related issues:
#12, #56, #61

# config
A new module (folder) to configure global settings, such as logger, DEBUG,  PATH_PSSM_SOURCE, and amino acid names and properties.

# generate
`settings.py`:
It's deleted. The global name "PATH_PSSM_SOURCE" can now be set with config module.

`DataGenerator.py`:
- **Bugs:** Molecule group attribute rotation 'center' was the center of interface but not the center of molecule. The augmentated molecule features are wrong if you have used 'add_feature' on your previous data. Then you have to fix them by updating the values of 'rotation' center.

`GridTools.py`:
- Remove class parameters 'hdf5_file' and 'logger'
- did not touch deprecated code 

# targets
- raise error when native complex not exist or is empty
- warn when target exsit in HDF5 and remove the old one.

`rmsd_fnat.py`:
 - Add this new helper module to calculate RMSDs and FNAT.

`dockQ.py`:
- Refacotr code to use helper module to calculate RMSDs and FNAT.

`binary_class.py`:
 - Using helper module to make binary_class independent from dockQ.py (it needs IRMSD generated from dockQ.py before).

# features
- raise ValueError if no interface contact found
- warn if interface contact is less than 5

`AtomicFeature.py`:
 - Remove parameters "root_export" and "individual_directory" which set paths for exporting data to plain files.
 - Uniform the key of atom info to (chainID, resSeq, resName, name) and that of residue to (chainID, resSeq, resName)

`FullPSSM.py` and `PSSM_IC.py`:
They can actually share one set of code. 
So FullPSSM class is refactored to include the computation of PSSM IC. 
Thus in PSSM_IC.py, just need to import FullPSSM and use it.

`ResidueDensity.py`:
 **Bugs:** WRONG residue type "LLE", changed to "ILE" (using config)  (!! All the hdf5 files genereted before have the wrong residue name in groups features/features_raw/mapped_features).

`FeatureClass.py`:
- Remove self.export_directories which is only used in AtomFeature.py
- Remove self.error which detect non interface contact.
   
# tools
`pdb2sql.py`:
- Uniform the key of atom info to (chainID, resSeq, resName, name) and that of residue to (chainID, resSeq, resName)

`StructureSimilarity.py`:
added import notes:
     1. The decoy and pdb must have consistent residue numbering.
     2. The zone files here are different with those from ProFit. The meaning of zone in our code:
               -  lzone: zone only for fitting.
               - izone: zone for both fitting and rmsd calculation.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -33,20 +33,20 @@ before_install:
...
+  # - conda install cython
+  # - sudo apt-get install -y libxml2 libxml2-dev libjson0 libjson0-dev
+  # - wget http://github.com/mittinatten/freesasa/releases/download/2.0.2/freesasa-2.0.2.tar.gz
+  # - mkdir freesasa
+  # - tar -xvf freesasa-2.0.2.tar.gz -C freesasa --strip-components=1
+  # - cd freesasa
+  # - ./configure --disable-xml --disable-json --enable-python-bindings --with-python=python3.6 CFLAGS="-fPIC" --prefix=$HOME
+  # - make
+  # - make install
+  # - cd ../
```
Comment (NicoRenaud) [inline .travis.yml]: if freesasa is pip installable remove all of that and put simply free sasa in the install requirements

Comment (CunliangGeng) [reply .travis.yml]: Yep, it is changed.

File: deeprank/tools/pdb2sql.py [hunk 2]
```diff
@@ -1,64 +1,70 @@
 import os
```
[the remark this answers is not in the document]
Comment (CunliangGeng) [reply deeprank/tools/pdb2sql.py]: Totally agree. Later I'll see how to merge some repeated code.

File: deeprank/tools/pdb2sql.py [hunk 3]
```diff
@@ -116,7 +126,7 @@ def _create_sql(self):
             'resName': [17, 20],
             'chainID': [21, 22],
             'resSeq': [22, 26],
-            'iCode': [26, 26],
+            'iCode': [26, 27],
```
[the remark this answers is not in the document]
Comment (CunliangGeng) [reply deeprank/tools/pdb2sql.py]: yes, iCode occupies the 27th column. then should set the index as [26:27].  [26:26] would return empty.

**Pull request merged**